onadata/apps/logger/management/commands/clean_duplicated_submissions.py

```diff
@@ -0,0 +1,121 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+from django.conf import settings
+from django.core.management.base import BaseCommand, CommandError
+from django.db import transaction
+from django.db.models.aggregates import Count
+from django.utils import timezone
+
+from onadata.apps.logger.models.instance import Instance
+from onadata.apps.logger.models.attachment import Attachment
+from onadata.apps.viewer.models.parsed_instance import ParsedInstance
+from onadata.libs.utils.common_tags import MONGO_STRFTIME
+
+
+class Command(BaseCommand):
+
+    help = "Deletes revisions (by chunks) for a given app [and model]"
+
+    def add_arguments(self, parser):
+        super(Command, self).add_arguments(parser)
+
+        parser.add_argument(
+            "--user",
+            default=None,
+            help="Specify a username to clean up only there forms",
+        )
+
+        parser.add_argument(
+            "--xform",
+            default=None,
+            help="Specify a XForm's `id_string to clean up only this form",
+        )
+
+        parser.add_argument(
+            "--delete",
+            action='store_true',
+            default=False,
+            help="Force delete instead of soft delete (i.e tagging instance as deleted). Default is False",
+        )
+
+    def handle(self, *app_labels, **options):
+        username = options['user']
+        xform_id_string = options['xform']
+        delete = options['delete']
+
+        # Delete revisions.
+        query = Instance.objects
+        if xform_id_string:
+            query = query.filter(xform__id_string=xform_id_string)
+
+        if username:
+            query = query.filter(xform__user__username=username)
+
+        query = query.values_list(
+            'uuid', flat=True
+        ).annotate(
+            count_uuid=Count('uuid')
+        ).filter(
+            count_uuid__gt=1
+        ).distinct()
+
+        for uuid in query.all():
```

Comment:
I was going to use it by chunk as we already did for other tasks. Because query only fetch `uuid` which is an indexed field, I thought that would be ok. 
To confirm my theory, I also run the query in a separate container on `OCHA`. It took 3 minutes to complete but it did run flawlessly.